### ISSUE TITLE
To extend CellFieldFromComposeExtended to CellBases 

### ISSUE THREAD

**Issue by fverdugo** [gridap/Gridap.jl#26]
The functionality provided by `CellFieldFromComposeExtended` has to be extended to support `CellBasis` objects.

For instance, this would be required to remove the commented part in line:
https://github.com/gridap/Gridap.jl/blob/c0777205d5a06da330911781a782c5c3e3bfd675/test/FESpacesTests/NonLinearFEOperatorsTests.jl#L53

Essentially, one needs to extend  `CellFieldFromComposeExtended` to `CellMapFromComposeExtended`, which in turn will require to extend `FieldFromComposeExtended` to `MapFromComposeExtended`.

**EDIT**

As a part of this re factoring, it would be useful that we can create `CellFields` giving more than one auxiliary cell field. (useful for Multi-field problems). E.g., 
```julia
νfun(x,u1,u2) = (u2*u1+1.0)*x[1]
ν(u1,u2) = CellField(trian,νfun,u1,u2)
```

**Comment by fverdugo** [comment]
Achtung! The design of `CellFieldFromComposeExtended` was wrong. It has been fixed in commit 98983e8. Previously, the `CellField` given in the 3rd argument was assumed to be evaluated in the physical space (i.e., in the range of the `CellGeomap` appearing in the 2nd argument). However, it has to be evaluated at the reference space since in practice it will be a FEFunciton... With this change, `CellFieldFromComposeExtended` is conceptually equivalent to a binary operation. Thus it can be removed in the future. The issue, is not yet closed since we want the same functionality for `CellBasis`. If we reimplement the `CellFieldFromComposeExtended` as a binary operation, the extension to `CellBasis` will be for free.

**Comment by fverdugo** [comment]
Partially solved this issue in commit 64ad4d4b17a9f9a59471414799ffe25a6084dfe7.

The fix for multifields is still to be done.

**Comment by fverdugo** [comment]
Fixed in commit 8164e97788a69b69b841e4b28db5775c15f538cc

Closing issue
